flink-connectors/flink-connector-aws-kinesis-data-streams/src/main/java/org/apache/flink/connector/kinesis/sink/KinesisDataStreamsSinkElementConverter.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.connector.kinesis.sink;
+
+import org.apache.flink.annotation.PublicEvolving;
+import org.apache.flink.api.common.serialization.SerializationSchema;
+import org.apache.flink.api.connector.sink.SinkWriter;
+import org.apache.flink.connector.base.sink.writer.ElementConverter;
+import org.apache.flink.util.Preconditions;
+
+import software.amazon.awssdk.core.SdkBytes;
+import software.amazon.awssdk.services.kinesis.model.PutRecordsRequestEntry;
+
+import java.io.Serializable;
+import java.util.function.Function;
+
+/**
+ * An implementation of the {@link ElementConverter} that uses the AWS v2 Kinesis SDK. The user only
+ * needs to provide a {@link SerializationSchema} of the {@code InputT} and a {@link
+ * PartitionKeyGenerator} lambda to transform the input element into a String.
+ */
+@PublicEvolving
+public class KinesisDataStreamsSinkElementConverter<InputT>
+        implements ElementConverter<InputT, PutRecordsRequestEntry> {
+
+    /** A serialization schema to specify how the input element should be serialized. */
+    private final SerializationSchema<InputT> serializationSchema;
+
+    /** A partition key generator lambda that produces a string from the input element. */
```

Comment:
nit: Does not need to be a lambda. I think it is a `FunctionalInterface`

Reply:
Cool, is the idea to modify the message here and add a `@FunctionalInterface` to the interface def?

Reply:
Done :-)